anagram.cpp

```diff
@@ -1,13 +1,51 @@
 #include "anagram.h"
+#include <iostream>
+#include <algorithm>
+using namespace std;
+
+bool Anagram::compareSortedStrings(string& word1, string& word2)
+{
+    for (int i = 0; i < word1.length(); i++)
+        if (word1[i] != word2[i])
+            return false;
+}
 
 bool Anagram::WordPairIsAnagram(const std::string& word1, const std::string& word2) {
     //Fill the correct implementation here
-    return false;
+    // Sorting both the strings 
+    string sorted1 = word1;
+    string sorted2 = word2;
```

Comment:
Clear preprocessing.
Notice that the erase-tolower-sort repeats for each word.
There is another example in Wikipedia: "Church of Scientology". Add a test for that. Before you fix anything, make a common pre-processing function to remove the repetition. Then fix it once.

Reply:
All suggested changes implemented. Please check. Thanks in advance!